superset/databases/commands/test_connection.py

```diff
@@ -55,25 +56,44 @@ def run(self) -> None:
                 impersonate_user=self._properties.get("impersonate_user", False),
                 encrypted_extra=self._properties.get("encrypted_extra", "{}"),
             )
-            if database is not None:
-                database.set_sqlalchemy_uri(uri)
-                database.db_engine_spec.mutate_db_for_connection_test(database)
-                username = self._actor.username if self._actor is not None else None
-                engine = database.get_sqla_engine(user_name=username)
+            if database is None:
+                raise DBAPIError("Database is not found", None, None)
```

Comment:
We can actually remove this and fix the type annotation in `DatabaseDAO.build_db_for_connection_test`, the class method never returns `None`.